src/local.rs

```diff
@@ -115,6 +125,38 @@ impl LocalAsset {
         }
     }
 
+    /// Creates a new directory, including all parent directories
+    pub fn create_directory(dest: &str) -> Result<PathBuf> {
+        let dest_path = PathBuf::from(dest);
+        match fs::create_dir_all(&dest_path) {
+            Ok(_) => Ok(dest_path),
+            Err(details) => Err(AxoassetError::LocalAssetWriteNewFailed {
```

Comment:
I would create a new error for the failure of dir creation (it will be helpful to signal the dif between an asset write failure and a dir write failure imo)